### PR TITLE
Update triggerSurvey to wait for authentication

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue:

- #5073

## Relevant technical choices

- Defers side-effect of fetching authentication if there is already a survey set
- Adds a Jest test that failed before the fix was implemented
- Adds simple E2E tests for extra coverage

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: tests/e2e/plugins/tracking-opt-in.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * @package   Google\Site_Kit
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+add_filter(
+	'get_user_option_googlesitekit_tracking_optin',
+	'__return_true'
+);
```
Comment (felixarntz) [inline tests/e2e/plugins/tracking-opt-in.php:18]: Why is this needed? I thought user surveys are being shown regardless, we only show a more explicit consent note on them when the user is _not_ opted in.

Comment (aaemnnosttv) [reply tests/e2e/plugins/tracking-opt-in.php:18]: Hmm, you're right. I misread this guard before when I was trying to get it to work.
https://github.com/google/site-kit-wp/blob/202d55f0a1f19f405a82fdfb8576e0ab02464080/assets/js/components/surveys/CurrentSurvey.js#L230-L237

I'll remove it as it's not needed 👍

**Review by felixarntz — APPROVED** [review]
Awesome work, especially with the e2e tests!
